src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -115,7 +115,25 @@ internal class DependencyResolver
             queries.Remove("displayProperty");
 
             // need to url decode uid from input content
-            var (error, resolvedHref, display, xrefSpec) = _xrefMap.Value.Resolve(Uri.UnescapeDataString(uid), href, displayProperty, text, declaringFile);
+            var (error, resolvedHref, xrefSpec) = _xrefMap.Value.Resolve(Uri.UnescapeDataString(uid), href);
+
+            var name = xrefSpec?.GetXrefPropertyValue("name");
+
+            // for internal UID, the display property can not be markdown or inline markdown
+            // because the display text should be plain text
+            string displayPropertyValue = null;
+            if (xrefSpec is InternalXrefSpec internalSpec
+                && !new JsonSchemaContentType[] { JsonSchemaContentType.Markdown, JsonSchemaContentType.InlineMarkdown }
```

Comment:
avoid the array allocation here using `contentType != Markdown && contentType != InlineMarkdown`